tests/CMakeLists.txt

```
@@ -39,39 +35,43 @@ else()
   message(WARNING "You are using an unsupported compiler! Compilation has only been tested with Clang and GCC, detected ${CMAKE_CXX_COMPILER_ID}.")
 endif()
 
+list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
+
 find_package(ut REQUIRED CONFIG)
 find_package(libhal REQUIRED CONFIG)
+find_package(libhal-util REQUIRED CONFIG)
 
 add_executable(${PROJECT_NAME}
-  empty.test.cpp
-  main.test.cpp
-)
+  
+  #Source
+  ../src/mpu6050.cpp
 
-target_include_directories(${PROJECT_NAME} PUBLIC . ../include)
+  #Tests
+  mpu.test.cpp
+  main.test.cpp)
+
+target_include_directories(${PROJECT_NAME} PUBLIC . ../include ../src)
 target_compile_options(${PROJECT_NAME} PRIVATE
   --coverage
   -fprofile-arcs
   -ftest-coverage
-  ${ASAN_FLAG}
   -Werror
   -Wall
   -Wextra
   -Wshadow
   -Wnon-virtual-dtor
   -Wno-gnu-statement-expression
-  -pedantic)
+  -pedantic
+  -fsanitize=address)
 target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_20)
+set_target_properties(${PROJECT_NAME} PROPERTIES CXX_EXTENSIONS OFF)
 target_link_options(${PROJECT_NAME} PRIVATE
   --coverage
   -fprofile-arcs
   -ftest-coverage
-  ${ASAN_FLAG})
-target_link_libraries(${PROJECT_NAME} PRIVATE boost-ext-ut::ut libhal::libhal)
-
-add_custom_command(TARGET ${PROJECT_NAME}
-  POST_BUILD
-  COMMAND ${CMAKE_COMMAND}
-  -P ${CMAKE_SOURCE_DIR}/remove_gcda.cmake
-  COMMENT "Deleting Previous Coverage Files..."
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-)
+  -fsanitize=address)
+target_link_libraries(${PROJECT_NAME} PRIVATE
+  boost-ext-ut::ut
+  libhal::libhal
+  libhal::util
+  libhal-mpu)
```

```suggestion
target_link_libraries(${PROJECT_NAME} PRIVATE
  boost-ext-ut::ut
  libhal::libhal
  libhal::util)
```

We actually want to remove the package because then it doesn't show up in coverage.